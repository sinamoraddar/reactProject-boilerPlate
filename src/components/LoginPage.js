import React from 'react';
import { connect } from 'react-redux';
import { startLogIn } from '../actions/auth';

export const LoginPage = ({ startLogIn }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">boilerplate</h1>
            <p>tag line for app</p>
            <button className="button" onClick={startLogIn}>Login with Google</button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogIn: () => dispatch(startLogIn())
})

export default connect(undefined, mapDispatchToProps)(LoginPage);