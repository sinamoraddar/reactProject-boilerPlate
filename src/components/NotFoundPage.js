import React from 'react';
import { Link } from 'react-router-dom';
const NotFoundPage = () => (
    <div>
        404 page was not found
        <Link to="/">Go home</Link >
    </div>
);
export default NotFoundPage;