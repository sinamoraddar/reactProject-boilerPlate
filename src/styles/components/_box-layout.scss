.box-layout {
  background: url("/images/background.jpg");
  background-size: cover;
  display: flex;
  width: 100vw;
  height: 100vh;
  justify-content: center;
  align-items: center;
}

.box-layout__box {
  background: fade-out(#ffffff, 0.25);
  height: 200px;
  border-radius: 3px;
  text-align: center;
  padding: $l-size $m-size;
  width: 25rem;
}

.box-layout__title {
  margin: 0 0 $m-size 0;
  line-height: 1;
  padding: 0;
}
