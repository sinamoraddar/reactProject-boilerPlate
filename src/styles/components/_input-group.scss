.input-group {
  display: flex;
  margin-bottom: $l-size;
}
.input-group__item {
  margin-right: $s-size;
}

@media screen and (max-width: $mobile-breakpoint) {
  .input-group {
    flex-direction: column;
    margin-bottom: $m-size;
  }
  
  .input-group__item {
    margin: 0 0 $s-size 0;
  }
}
