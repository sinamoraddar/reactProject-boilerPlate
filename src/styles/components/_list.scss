.list-header {
  background: $off-white;
  border: 1px solid darken($off-white, 7%);
  color: $less-darker-gray;
  display: flex;
  justify-content: space-between;
  padding: $s-size $m-size;
}

.list-body {
  margin-bottom: $m-size;

  @media screen and(min-width:$mobile-breakpoint) {
    margin-bottom: $l-size;
  }
}

.list-item {
  border: 1px solid darken($off-white, 7%);
  border-top: none;
  color: $dark-gray;
  display: flex;
  flex-direction: column;
  padding: $s-size;
  text-decoration: none;
  transition: background 0.3s ease;
  &:hover {
    background: $off-white;
  }

  @media screen and (min-width: $mobile-breakpoint) {
    align-items: center;
    flex-direction: row;
    justify-content: space-between;
    padding: $m-size;
  }
}

.list-item--message {
  align-items: center;
  color: $gray;
  justify-content: center;
  padding: $m-size;
  &:hover {
    background: initial;
  }
}

.list-item__title {
  margin: 0;
  word-break: break-all;
}

.list-item__sub-title {
  color: $less-darker-gray;
  font-size: $font-size-small;
}

.list-item__data {
  margin: $s-size 0 0 0;
  @media screen and(min-width:$mobile-breakpoint) {
    margin: 0;
    padding-left: $s-size;
  }
}
