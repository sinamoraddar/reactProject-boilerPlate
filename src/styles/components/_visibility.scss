@media screen and(min-width:$mobile-breakpoint) {
  .show-for-mobile {
    display: none;
  }
}

@media screen and(max-width:$mobile-breakpoint - .01rem) {
  .show-for-desktop {
    display: none;
  }
}
