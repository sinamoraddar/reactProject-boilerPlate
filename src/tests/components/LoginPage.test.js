import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('should render LoginPage component', () => {
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
});


test('should call startLogIn on button click', () => {
    const startLogIn = jest.fn();
    const wrapper = shallow(<LoginPage startLogIn={startLogIn} />);
    wrapper.find('button').simulate('click');
    expect(startLogIn).toHaveBeenCalled();
});