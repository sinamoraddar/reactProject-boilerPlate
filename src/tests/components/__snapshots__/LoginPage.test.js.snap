// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage component 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      boilerplate
    </h1>
    <p>
      tag line for app
    </p>
    <button
      className="button"
    >
      Login with Google
    </button>
  </div>
</div>
`;
